components/scream/src/share/scream_types.hpp

```diff
@@ -37,6 +37,14 @@ enum class RunType {
   Restarted
 };
 
+// An enum for specifying fields data type
+enum class DataType {
```

Comment:
@bartgol, what's this? Is this left over from your original `_t` enum type idea? Seems like we could start from this and change if we think of something better.

Reply:
Yep, that's it. I forgot about this. I am not fond of `_t` since my std-mind set parses that as a typename, while this is an enum value. Some other options:

- `I`, `F`, `D`: maybe not clear enough
- `int_type`, `float_type`, `double_type`: not too different from `_t` (`blah_type` is often a typename, not a value), but maybe a bit better
- `int_data`, `float_data`, `double_data`: like the above, but less ambiguity with typenames.
- `type_int`, `type_float`, `type_double`: flipping the words seems less confusing?
- `IntType`, `FloatType`, `DoubleType`: no conflicts with existing typenames.

Maybe the last one is my favorite. But there may be better options.

Reply:
I like the last one too. Why don't we go with that? If we need to further "scope" it, we can always make it an anonymous enum within the `Field` class itself (but no need this PR).

Reply:
Being an enum class, I think it's already scoped enough.